### ISSUE TITLE
LaTeXStrings in axis labels

### ISSUE THREAD

**Issue by grahamas** [JuliaPlots/Makie.jl#730]
Presently, you get errors, different ones depending on whether you go through MakieLayout or through Makie.

I think this is different from #264 which seems to request equations not in the labels, but within the plot itself, and indeed that's what JuliaPlots/MakieTeX.jl provides.

```julia
using LaTeXStrings, Makie
scene = Scene(); plot!(scene, 1:4, 1:4); xlabel!(scene, L"a_t")
```
gives:
```julia
ERROR: MethodError: no method matching push!(::Annotations{...}, ::LaTeXString, ::Tuple{Float64,Float64}; textsize=0.20936732590198515, align=(:center, :top), rotation=0.0, color=:black, font="Dejavu Sans")
Closest candidates are:
  push!(::Any, ::Any, ::Any) at abstractarray.jl:2252 got unsupported keyword arguments "textsize", "align", "rotation", "color", "font"
  push!(::Any, ::Any, ::Any, ::Any...) at abstractarray.jl:2253 got unsupported keyword arguments "textsize", "align", "rotation", "color", "font"
  push!(::OffsetArrays.OffsetArray{T,1,AA} where AA<:AbstractArray{T,1} where T, ::Any...) at /home/graham/.julia/packages/OffsetArrays/sUnpU/src/OffsetArrays.jl:337 got unsupported keyword arguments "textsize", "align", "rotation", "color", "font"
  ...
Stacktrace:
 [1] draw_titles(::Annotations{...}, ::Base.Iterators.Zip{Tuple{Array{Float64,1},Array{String,1}}}, ::Base.Iterators.Zip{Tuple{Array{Float64,1},Array{String,1}}}, ::Tuple{Float64,Float64}, ::Tuple{Float64,Float64}, ::Tuple{String,String}, ::Tuple{Float64,Float64}, ::Tuple{Float64,Float64}, ::Tuple{Float64,Float64}, ::Tuple{LaTeXString,String}, ::Tuple{Symbol,Symbol}, ::Tuple{Float64,Float64}, ::Tuple{Float64,Float64}, ::Tuple{Tuple{Symbol,Symbol},Tuple{Symbol,Symbol}}, ::Tuple{String,String}, ::Nothing) at /home/graham/.julia/packages/AbstractPlotting/tgvKi/src/basic_recipes/axis.jl:407
 [2] draw_axis2d(::Annotations{...}, ::LineSegments{...}, ::Tuple{LineSegments{...},LineSegments{...}}, ::Tuple{LineSegments{...},LineSegments{...}}, ::StaticArrays.SArray{Tuple{4,4},Float32,2,16}, ::Float64, ::Tuple{Tuple{Float32,Float32},Tuple{Float32,Float32}}, ::Tuple{Tuple{Array{Float64,1},Array{Float64,1}},Tuple{Array{String,1},Array{String,1}}}, ::Tuple{Bool,Bool}, ::Tuple{Bool,Bool}, ::Tuple{Bool,Bool}, ::Tuple{Float64,Float64}, ::Tuple{Tuple{Symbol,Float64},Tuple{Symbol,Float64}}, ::Tuple{Nothing,Nothing}, ::Tuple{Int64,Int64}, ::Tuple{Tuple{Symbol,Float64},Tuple{Symbol,Float64}}, ::Tuple{Nothing,Nothing}, ::Tuple{Symbol,Symbol}, ::Tuple{Int64,Int64}, ::Tuple{Float64,Float64}, ::Tuple{Tuple{Symbol,Symbol},Tuple{Symbol,Symbol}}, ::Tuple{String,String}, ::Int64, ::Int64, ::Tuple{Int64,Int64}, ::Tuple{Tuple{Symbol,Float64},Tuple{Symbol,Float64}}, ::Tuple{Nothing,Nothing}, ::Tuple{Float64,Float64}, ::Float64, ::Symbol, ::Nothing, ::Nothing, ::Bool, ::Float64, ::Tuple{Tuple{Bool,Bool},Tuple{Bool,Bool}}, ::Tuple{LaTeXString,String}, ::Tuple{Symbol,Symbol}, ::Tuple{Int64,Int64}, ::Tuple{Float64,Float64}, ::Tuple{Tuple{Symbol,Symbol},Tuple{Symbol,Symbol}}, ::Tuple{String,String}, ::Nothing) at /home/graham/.julia/packages/AbstractPlotting/tgvKi/src/basic_recipes/axis.jl:524
 [3] (::Observables.MapUpdater{typeof(AbstractPlotting.draw_axis2d),Nothing})(::Annotations{...}, ::Vararg{Any,N} where N) at /home/graham/.julia/packages/Observables/A7I8T/src/Observables.jl:326
 [4] (::Observables.OnUpdate{Observables.MapUpdater{typeof(AbstractPlotting.draw_axis2d),Nothing},Tuple{Observable{Annotations{...}},Observable{LineSegments{...}},Observable{Tuple{LineSegments{...},LineSegments{...}}},Observable{Tuple{LineSegments{...},LineSegments{...}}},Observable{StaticArrays.SArray{Tuple{4,4},Float32,2,16}},Observable{Any},Observable{Tuple{Tuple{Float32,Float32},Tuple{Float32,Float32}}},Observable{Any},Observable{Tuple{Bool,Bool}},Observable{Tuple{Bool,Bool}},Observable{Tuple{Bool,Bool}},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any}}})(::Tuple{LaTeXString,String}) at /home/graham/.julia/packages/Observables/A7I8T/src/Observables.jl:293
 [5] setindex!(::Observable{Any}, ::Tuple{LaTeXString,String}; notify::Observables.var"#9#11") at /home/graham/.julia/packages/Observables/A7I8T/src/Observables.jl:205
 [6] setindex!(::Observable{Any}, ::Tuple{LaTeXString,String}) at /home/graham/.julia/packages/Observables/A7I8T/src/Observables.jl:201
 [7] xlabel!(::Scene, ::LaTeXString) at /home/graham/.julia/packages/AbstractPlotting/tgvKi/src/shorthands.jl:10
 [8] top-level scope at REPL[4]:1
```
and then with MakieLayout:
```julia
using LaTeXStrings, AbstractPlotting.MakieLayout
scene, layout = layoutscene(); layout[1,1] = ax = LAxis(scene, xlabel=L"a_t")
```
gives
```julia
ERROR: ArgumentError: regex matching is only available for the String type; use String(s) to convert
Stacktrace:
 [1] match(::Regex, ::LaTeXString, ::Int64) at ./regex.jl:295
 [2] match(::Regex, ::LaTeXString) at ./regex.jl:294
 [3] iswhitespace(::LaTeXString) at /home/graham/.julia/packages/AbstractPlotting/tgvKi/src/makielayout/lineaxis.jl:296
 [4] (::AbstractPlotting.MakieLayout.var"#130#151"{Observable{Tuple{Float32,Tuple{Float32,Float32},Bool}},Text{...}})(::Bool, ::LaTeXString, ::Bool, ::Float32, ::Float32, ::Float32, ::Float32, ::Bool, ::Float32, ::Float32, ::String, ::String) at /home/graham/.julia/packages/AbstractPlotting/tgvKi/src/makielayout/lineaxis.jl:253
 [5] lift(::Function, ::Observable{Any}, ::Observable{Any}, ::Vararg{Any,N} where N) at /home/graham/.julia/packages/AbstractPlotting/tgvKi/src/interaction/nodes.jl:15
 [6] AbstractPlotting.MakieLayout.LineAxis(::Scene; kwargs::Base.Iterators.Pairs{Symbol,Observable,NTuple{28,Symbol},NamedTuple{(:endpoints, :limits, :flipped, :ticklabelrotation, :ticklabelalign, :labelsize, :labelpadding, :ticklabelpad, :labelvisible, :label, :labelfont, :ticklabelfont, :ticklabelcolor, :labelcolor, :tickalign, :ticklabelspace, :ticks, :tickformat, :ticklabelsvisible, :ticksvisible, :spinevisible, :spinecolor, :spinewidth, :ticklabelsize, :trimspine, :ticksize, :reversed, :tickwidth),Tuple{Observable{Tuple{Point{2,Float32},Point{2,Float32}}},Observable{Tuple{Float32,Float32}},Observable{Bool},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Bool},Observable{Symbol},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any},Observable{Any}}}}) at /home/graham/.julia/packages/AbstractPlotting/tgvKi/src/makielayout/lineaxis.jl:247
 [7] LAxis(::Scene; bbox::Nothing, kwargs::Base.Iterators.Pairs{Symbol,LaTeXString,Tuple{Symbol},NamedTuple{(:xlabel,),Tuple{LaTeXString}}}) at /home/graham/.julia/packages/AbstractPlotting/tgvKi/src/makielayout/lobjects/laxis.jl:149
 [8] top-level scope at REPL[10]:1
```

**Comment by ederag** [comment]
At least the following allows to display simple text and ticks labels (decimal numbers):
```julia
using FreeTypeAbstraction
Figure(...,
    font=FreeTypeAbstraction.findfont("cmr10")
)
```
Log10 scale superscripts and greek characters currently require an external tool.

Also tried `cmunrm.ttf` and the largest `cmunsl.ttf` from
https://sourceforge.net/projects/cm-unicode/files/cm-unicode/0.7.0/cm-unicode-0.7.0-ttf.tar.xz
but the greek letters are not math style, and superscripts are still missing.

So it is great that there are [plans](https://discourse.julialang.org/t/formatting-plots-with-makie/53034/7) to integrate some LaTeX into Makie (awesome work so far !).

**Comment by fbanning** [comment]
Fixed in Makie v0.15
